src/plugins/layout/shadowdom.ts

```diff
@@ -272,25 +208,93 @@ export class ShadowDom {
       this.removeAllClasses(next);
     }
 
-    let removed = this.removedNodes.childNodes;
-    for (let i = 0; i < removed.length; i++) {
-      let next = removed[i] as IShadowDomNode;
-      if (!this.hasClass(next, NewNodeClassName)) {
-        summary.removedNodes.push(next);
-      }
+    let removedNodes = this.removedNodes.childNodes;
+    for (let i = 0; i < removedNodes.length; i++) {
+      traverseNodeTree(removedNodes[i], (shadowNode: IShadowDomNode) => {
+        if (this.hasClass(shadowNode, NewNodeClassName)) {
+          delete shadowNode.node[NodeIndex];
+        } else if (this.hasClass(shadowNode, MovedNodeClassName)) {
+          summary.removedNodes.push(shadowNode);
+        }
+      });
     }
 
     return summary;
   }
 
+  public createDomIndexJson(): object {
+
+    function writeIndex(node: Node, json: object) {
```

Comment:
Created NumberJson type, however the compiler doesn't seem to enforce numeric keys. Still looks better for code readability though, thanks.

---
In reply to: [131239582](https://github.com/Microsoft/clarity-js/pull/25#discussion_r131239582) [](ancestors = 131239582)